### PR TITLE
expanded await task output variables

### PR DESCRIPTION
**Pull request by benPearce1**
Added link to all awaited tasks
Added polling logs for task state

image

Added output variables:
`completed_successfully` : overall status of tasks (false if any fail)

`<environment name | tenant name>.completed_successfully`: one completed successfully flag per task, with tenant or environment context in the variable name, spaces replaced with underscores. e.g 
- `dev_2.completed_successfully`
- `tenant_1.completed_successfully`

`server_task_results`: JSON representation of all tasks with outcome. E.g:

```json
[
    {
        "serverTaskId": "ServerTasks-135172",
        "environmentName": "dev",
        "successful": true
    },
    {
        "serverTaskId": "ServerTasks-135173",
        "environmentName": "dev 2",
        "successful": false
    }
]
```

```json
[
    {
        "serverTaskId": "ServerTasks-135176",
        "tenantName": "tenant 4",
        "successful": true
    },
    {
        "serverTaskId": "ServerTasks-135175",
        "tenantName": "tenant 2",
        "successful": true
    },
    {
        "serverTaskId": "ServerTasks-135174",
        "tenantName": "tenant 1",
        "successful": true
    }
]
```

Polling messages:
`Deployment to environment 'dev 2' is 'Executing'`
`Deployment to environment 'test' for tenant 'tenant-EB7CC' is 'Executing'`
`Runbook run to environment 'dev' is 'Executing'`

### CODE REVIEW

File: source/tasks/AwaitTask/AwaitTaskV6/waiter.ts [hunk 1]
```diff
@@ -27,7 +34,7 @@ export class Waiter {
         const waiter = new ServerTaskWaiter(client, inputParameters.space);
 
         const taskIds: string[] = [];
-        const failedTaskIds: string[] = [];
+        const deploymentResults: DeploymentResult[] = [];
```
Comment (slewis74) [inline source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:37]: I'd missed this in the earlier reviews. This isn't always a `DeploymentResult`, sometimes it's a `RunbookRunResult`.

File: source/tasks/AwaitTask/AwaitTaskV6/waiter.ts [hunk 2]
```diff
@@ -37,26 +44,60 @@ export class Waiter {
...
+                if (deploymentResult?.tenantName) {
+                    context += ` for tenant '${deploymentResult?.tenantName}'`;
+                }
 
-            if (t.IsCompleted) {
-                this.logger.info?.(`Deployment${context} ${t.State === TaskState.Success ? "completed successfully" : "did not complete successfully"}`);
-            } else {
-                this.logger.info?.(`Deployment${context} is '${t.State}'`);
+                if (t.IsCompleted) {
+                    this.logger.info?.(`Deployment${context} ${t.State === TaskState.Success ? "completed successfully" : "did not complete successfully"}`);
```
Comment (slewis74) [inline source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:56]: These are tasks. Could be a deployment, but could also be a Runbook Run. I think the links below will still work the same, but the text here will be misleading.

Comment (benPearce1) [reply source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:56]: Deployment => Execution?

**Review by slewis74 — CHANGES_REQUESTED** [review]
Think we need to check the wording of some of the outputs, given the tasks could be runbook runs.

File: source/tasks/AwaitTask/AwaitTaskV6/waiter.ts (hunk 1 above)
Comment (slewis74) [reply source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:37]: maybe rename to `TaskResult`?

File: source/tasks/AwaitTask/AwaitTaskV6/waiter.ts (hunk 2 above)
Comment (slewis74) [reply source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:56]: yeah, we've been using that as the generic term to mean either of those things 👍

Comment (benPearce1) [reply source/tasks/AwaitTask/AwaitTaskV6/waiter.ts:56]: Polling messages updated, task type (deployment or runbook run) is provided by the task:
`Deployment to environment 'dev 2' is 'Executing'`
`Deployment to environment 'test' for tenant 'tenant-EB7CC' is 'Executing'`
`Runbook run to environment 'dev' is 'Executing'`

**Review by slewis74 — APPROVED** [review]

**Pull request merged**